aptible/resource_environment.go

```diff
@@ -47,9 +50,41 @@ func resourceEnvironment() *schema.Resource {
 func resourceEnvironmentCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
 	client := meta.(*aptible.Client)
 	handle := d.Get("handle").(string)
-	orgID := d.Get("org_id").(string)
 	stackID := int64(d.Get("stack_id").(int))
 
+	// there are a few  scenarios where org_id can be fully gathered from the data provided above
+	// 1. If it is provided explicitly
+	// 2. It is using a dedicated stack, and so when creating an environment, use the stack id to get stack
+	//    and check that stack's org id and use it
+	// 3. The user only belongs to one organization, so fall back to that. If they belong to multiple organizations
+	// 	  they will have to specify it explicitly at this point (if these points no longer work for inference.)
+	orgID := d.Get("org_id").(string) // scenario #1 outlined above
+	if orgID == "" {
+		stack, err := client.GetStack(stackID) // scenario #2 outlined above
+		if err != nil {
+			log.Println("There was an error trying to retrieve the stack with the stack id provided to determine"+
+				"an organization id.\n[Error] - ", err)
+			return generateDiagnosticsFromClientError(err)
+		}
```

Comment:
What do you think about not returning here since we still may be able to determine the org ID further down? Instead we could just log the client error and move on.

Reply:
Probably. I'm fine with that.